client/beefy/src/gossip.rs

```diff
@@ -193,8 +197,12 @@ where
 	fn message_allowed<'a>(
 		&'a self,
 	) -> Box<dyn FnMut(&PeerId, MessageIntent, &B::Hash, &[u8]) -> bool + 'a> {
-		let do_rebroadcast = {
+		let do_rebroadcast = || {
 			let now = Instant::now();
+			// TODO: right now we're using an object-level rebroadcast gate that will only
+			// allow **a single message** being rebroadcast every `REBROADCAST_AFTER` minutes.
+			//
+			// Should we instead have a per-message/hash rebroadcast cooldown?
```

Comment:
IMO we should not make this a closure. The way the gossip state machine currently works is that it will call into `message_allowed` to get a function for validation. When it's time to do a rebroadcast it will pass the returned function through all messages in order to check whether to (re-)propagate. If this isn't a closure then `do_rebroadcast` will have the same value for all messages and thus the comment above won't apply (i.e. we'll rebroadcast all messages since `do_rebroadcast` will stay true).

Reply:
ok, that makes sense